sklearn/metrics/tests/test_pairwise.py

```diff
@@ -407,6 +418,39 @@ def test_euclidean_distances():
     assert_greater(np.max(np.abs(wrong_D - D1)), .01)
 
 
+def test_masked_euclidean_distances():
+    # first check that we get right answer with missing values for X
+    X = np.array([[1.,   5.,   7.,   5.,  10.],
+                  [8., 2., 4., np.nan, 8.],
+                  [5., np.nan, 5., np.nan, 1.],
+                  [8., np.nan, np.nan, np.nan, np.nan]])
+    D1 = masked_euclidean_distances(X, missing_values="NaN")
+
+    D2 = np.array([[0.,   9.42072184,  12.97433364,  15.65247584],
+                  [9.42072184,   0.,   9.91631652,   0.],
+                  [12.97433364,   9.91631652,   0.,   6.70820393],
+                  [15.65247584,   0.,   6.70820393,   0.]])
+
+    assert_array_almost_equal(D1, D2)
+
+    # check with pairs of matrices with missing values
+    X = np.array([[1., np.nan, 3., 4., 2.],
+                  [np.nan, 4., 6., 1., np.nan],
+                  [3., np.nan, np.nan, np.nan, 1.]])
+
+    Y = np.array([[np.nan, 7., 7., np.nan, 2.],
+                  [np.nan, np.nan, 5., 4., 7.],
+                  [np.nan, np.nan, np.nan, 4., 5.]])
+
+    D3 = np.array([[6.32455532, 6.95221787, 4.74341649],
```

Comment:
I'd rather see the tests show working, at least in some cases. Certainly you should test the distance calculation in the `squared=True` case for readability, then test the invariance that `squared` is meant to obtain.
```
    assert_almost_equal(masked_euclidean_distances(X[:1], Y[:1], squared=True),
                        [[5/2 * ((7-3)**2 + (2-2)**2)]])
```